src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs

```diff
@@ -114,7 +119,7 @@ public static SslStreamCertificateContext Create(X509Certificate2 target, X509Ce
             return ctx;
         }
 
-        partial void AddRootCertificate(X509Certificate2? rootCertificate);
+        private partial void AddRootCertificate(X509Certificate2? rootCertificate);
```

Comment:
We could get rid of the partials in the other implementations by changing this to have a ref bool and the call to be something like

```C#
bool transferred = false;
ctx.AddRootCertificate(root, ref transferred);

if (!transferred)
{
    root.Dispose();
}
```

And that returns the feature to a strict opt-in (and the if should const-fold away)

Reply:
TIL `ref` works on partials. Makes sense.